modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/BinarySpatialOpTypeBinding.java

```diff
@@ -55,6 +62,18 @@ public Class getType() {
     
     @Override
     public Object getProperty(Object object, QName name) throws Exception {
-        return FESParseEncodeUtil.getProperty((BinarySpatialOperator) object, name);
+        Expression e = FESParseEncodeUtil.getProperty((BinarySpatialOperator) object, name);
+        if (e instanceof Literal && ((Literal) e).getValue() instanceof Geometry) {
+            return new EncoderDelegate() {
+                @Override
+                public void encode(ContentHandler output) throws Exception {                    
+                    Encoder encoder = new Encoder(new org.geotools.gml3.v3_2.GMLConfiguration());
```

Comment:
@NielsCharlier I think it is perfectly fine for wfs-ng to decide to make all its WFS 2.0 / FES 2.0 requests using GML 3.2.1, which is guaranteed to be supported. The problem is that this module may be used for other purposes. The API is lacking. There should probably be some sort of factory or configuration being injected into this binding. Looks at ```GML3EncodingUntils``` for example. I agree that GML 3.2.1 is likely a good choice, but it just seems wrong to hard-code this choice here.